### PR TITLE
Inclusion of Max Cap Requirement

### PR DESCRIPTION
**Pull request by sambuddhac**
This PR attempts to take into account the Max Capacity constraint in addition to the min cap constraint

### CODE REVIEW

**Comment by sambuddhac** [comment]
Draft PR in the process. Will change the status once I finalize the PR and let all know.

**Review by cfe316 — CHANGES_REQUESTED** [review]
Hi Sam,

Thanks for making this PR and tagging me as a reviewer.
This doesn't look like a complete feature. If I search for MinCapReq and MinCapRes I see that they're in all these files, so more changed will be necessary. I suppose in principle the rest of the changes could be made later.

```
write_outputs/write_outputs.jl
write_outputs/minimum_capacity_requirement/write_minimum_capacity_requirement.jl
write_outputs/write_subsidy_revenue.jl
write_outputs/write_net_revenue.jl
configure_settings/configure_settings.jl
model/generate_model.jl
model/core/discharge/investment_discharge.jl
model/policies/minimum_capacity_requirement.jl
load_inputs/load_inputs.jl
load_inputs/load_minimum_capacity_requirement.jl
```
I'm also not certain that this will work, technically, since line 191 adds to an expression which does not already exist.

As a stylistic matter, `:][!, R_ID]` should be shortened to `:R_ID]`.

**Comment by cfe316** [comment]
Ah, sorry, I hadn't seen that this was a draft PR.

**Comment by sambuddhac** [comment]
@cfe316 I will finish this and then finalize the PR. Will let y'all know.

**Comment by cfe316** [comment]
`MinCapReq` is mentioned in `write_net_revenue.jl`. Does this needs to be addressed in some similar way?

**Comment by sambuddhac** [comment]
This is now ready for review. @cfe316 @JesseJenkins @dharik13 , please let me know if there needs to be further modifications or if everything looks alright, then if it can be merged with `Develop`

**Review by cfe316 — APPROVED** [review]

File: src/write_outputs/write_outputs.jl [hunk 1]
```diff
@@ -178,6 +178,10 @@ function write_outputs(EP::Model, path::AbstractString, setup::Dict, inputs::Dic
 			dfMinCapReq = write_minimum_capacity_requirement(path, inputs, setup, EP)
 		end
 
+		if setup["MaxCapReq"] == 1 && has_duals(EP) == 1
+			dfMinCapReq = write_maximum_capacity_requirement(path, inputs, setup, EP)
```
Comment (cfe316) [inline src/write_outputs/write_outputs.jl]: This dataframe should be called dfMaxCapReq.

**Review by cfe316 — APPROVED** [review]

**Pull request merged**